stdlib/builtins.pyi

```diff
@@ -109,19 +111,19 @@ class object:
     def __dir__(self) -> Iterable[str]: ...
     def __init_subclass__(cls) -> None: ...
 
-class staticmethod(object):  # Special, only valid as a decorator.
-    __func__: Callable[..., Any]
+class staticmethod(Generic[_FuncT]):  # Special, only valid as a decorator.
+    __func__: _FuncT
     __isabstractmethod__: bool
-    def __init__(self, f: Callable[..., Any]) -> None: ...
+    def __init__(self, __f: _FuncT) -> None: ...
     def __new__(cls: Type[_T], *args: Any, **kwargs: Any) -> _T: ...
-    def __get__(self, obj: _T, type: Optional[Type[_T]] = ...) -> Callable[..., Any]: ...
+    def __get__(self, obj: _T, type: Type[_T] | None = ...) -> _FuncT: ...
```

Comment:
Static methods can be looked up as `TheClass.method`, and in that case there is no associated instance:

```suggestion
    def __get__(self, obj: _T | None, type: Type[_T] | None = ...) -> _FuncT: ...
```